java/arcs/core/analysis/PredicateUtils.kt

```diff
@@ -15,62 +15,79 @@ import arcs.core.data.InformationFlowLabel
 import arcs.core.data.InformationFlowLabel.Predicate
 import java.util.BitSet
 
-fun BitSet.toString(transform: ((Int) -> String)?): String {
-    if (transform == null) return "$this"
-    var labels = mutableListOf<String>()
-    var nextBit = nextSetBit(0)
-    while (nextBit != -1) {
-        labels.add(transform(nextBit))
-        if (nextBit == Int.MAX_VALUE) break
-        nextBit = nextSetBit(nextBit + 1)
-    }
-    return labels.joinToString(prefix = "{", postfix = "}")
-}
-
 /**
- * Returns a set of [BitSet] representation for the predicate using the given [indices] to
- * determine the index of an [InformationFlowLabel] in the bitset.
+ * Returns the predicate in Disjunctive Normal Form (DNF) as a set of [BitSet] pairs.
+ *
+ * The first [BitSet] in the pair is a mask that determines the set of valid bits in the second
+ * [BitSet]. The [indices] map is used to determine the index of an [InformationFlowLabel] in the
+ * bitset. Suppose that the universe of labels is {A, B, C}. Here are some examples:
+ *
+ *          Predicate : {(mask, bits)}
+ *                  A : {(100, 100)}
+ *             not(A) : {(100, 000)}
+ *            A and B : {(110, 110)}
+ *       A and not(B) : {(110, 100)}
+ *             A or B : {(100, 100), (010, 010)}
+ *        A or not(B) : {(100, 100), (010, 000)}
+ *   (A and B) or (C) : {(110, 110), (001, 001)}
  */
-fun InformationFlowLabel.Predicate.asSetOfBitSets(
+fun InformationFlowLabel.Predicate.asDNF(
     indices: Map<InformationFlowLabel, Int>
-): Set<BitSet> {
+): Set<Pair<BitSet, BitSet>> {
     // TODO(b/157530728): This is not very efficient. We will want to replace this with
     // an implementation that uses a Binary Decision Diagram (BDD).
     when (this) {
         is Predicate.Label -> {
-            val result = BitSet(indices.size)
-            result.set(requireNotNull(indices[label]))
-            return setOf(result)
+            val index = requireNotNull(indices[label])
```

Comment:
If you aren't expecting null values, you've better off using `indices.getValue(label)` which will return a nonnull result or throw if the value is missing.

Reply:
Done.